src/Components/Form/ColorForm.jsx

```diff
@@ -0,0 +1,52 @@
+import "./ColorForm.css";
+import { uid } from "uid";
+import ColorInput from "./ColorInput";
+
+export default function ColorForm({
+  onAddColor,
+  initialData = { role: "some color", hex: "#123456", contrastText: "#ffffff" },
+}) {
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    const newColor = {
+      id: uid(),
+      role: data.role || initialData.role,
+      hex: data.hex || initialData.hex,
+      contrastText: data.contrastText || initialData.contrastText,
+    };
+    event.target.elements.role.focus();
+    // console.log("object data:", event.target.elements.role);
+    onAddColor(newColor);
+    event.target.reset();
+  }
+
+  return (
+    <form className="input-form" onSubmit={handleSubmit}>
+      <label htmlFor="role">Role</label>
+      <input
+        className="input-field"
+        type="text"
+        id="role"
+        name="role"
+        placeholder={initialData.role}
+      />
+      <label htmlFor="hex">Hex</label>
+      <ColorInput
+        className="input-field"
```

Comment:
your Component does not receive a prop called "className" therefore this has no effect. Custom Components don't inherit the "standard" props that react components have like "id" or "className" or "onClick" and need to be handled manually. This means, that a component does not get the class automatically by giving it a prop className. Inside the component, you need to accept this prop and pass it to a standard component like a div, button or article element.

Reply:
ok, understood. Thank you.